### PR TITLE
fixes: resolving bugs related to edit/upload stories pages

### LINKED ISSUE #329: Mark mandatory fields in Add/Edit Story flow
# Description
All mandatory fields that are in the Add/Edit Story flow should have an asterisk on the title INSTEAD OF the input field.
See screenshot below:
Screen Shot 2021-04-27 at 10 20 40 PM

# Steps to Reproduce
1. Log in
2. Add New Story

# Notes:
Here's how it should look like:
Screen Shot 2021-04-27 at 10 21 41 PM

### LINKED ISSUE #331: Upload/Edit Story flow Story Year should have 2021 by default
# Description
When filling out the field "Story Year" the year should be pre-populated. Instead of there being an empty field, it should already be filled with the year "2021", if they want to change the year, the user can then click on the field and select something from the drop-down.

# Steps to Reproduce
1. Log In
2. Click on Upload Story
3. Scroll down to the Story Year

Current behaviour: the field is blank
Expected behaviour: the field is prepopulated with the year 2021

### LINKED ISSUE #332: Tags are one word only...
# Description
The tags are currently one word only. When the user presses space a new tag is created. This should not happen. **Space should not create a new tag. Only enter should create a new tag.**

# Steps to Reproduce
1. Add A story
2. Add Tags
3. Add a tag that does not exist and is a two-word item: for example "blue whale". Do this by typing "blue", space, "whale", enter

Current behaviour: Two tags are created "blue" and "whale"
Expected behaviour: One tag "blue whale"

### LINKED ISSUE #340: Favicon in the Header
Add the shoe project logo to the header. Currently its a default icon:

image

### PR DESCRIPTION
**Pull request by DinuWije**
## Major Changes:
- added favicon site-wide (closes #340)
- made it possible to add tags that are more than one word long (closes #332)
- year field in Upload Story page defaults to current year (closes #331)
- change how mandatory fields are indicated in the upload/edit story components (closes #329)
- MVP: Loosely followed design prompts from #321 (some more work needs to be done to match the designs exactly with Figma, but the needed functionality is there)

### CODE REVIEW

File: ui/src/pages/Admin/Edit/UploadStory.tsx [hunk 1]
```diff
@@ -221,6 +221,7 @@ export const UploadStory: React.FC<StoryProps> = ({
   const history = useHistory();
 
   const startYear = new Date().getFullYear();
+  // const prefilledYear = id ? currentStory.year : startYear.toString();
```
Comment (AbhiPrasad) [inline ui/src/pages/Admin/Edit/UploadStory.tsx]: Can we 🔥?

File: ui/src/pages/Admin/Edit/UploadStory.tsx [hunk 2]
```diff
@@ -568,38 +602,35 @@ export const UploadStory: React.FC<StoryProps> = ({
           <FormControl>
             <StyledBackgroundColor>
               <UploadStoriesHeading>Story Information</UploadStoriesHeading>
-              <UploadLabelsText>Title</UploadLabelsText>
+              <UploadLabelsText>Title*</UploadLabelsText>
```
Comment (AbhiPrasad) [inline ui/src/pages/Admin/Edit/UploadStory.tsx:604]: Why we adding *?

File: ui/src/pages/Admin/Edit/UploadStory.tsx [hunk 3]
```diff
@@ -748,8 +777,15 @@ export const UploadStory: React.FC<StoryProps> = ({
                   id="tags-outlined"
                   name="tags"
                   freeSolo
-                  options={tagOptions ? tagOptions : [""]}
+                  forcePopupIcon
+                  options={tagOptions || [""]}
```
Comment (AbhiPrasad) [inline ui/src/pages/Admin/Edit/UploadStory.tsx]: Why default to `[""]` vs. an empty array `[]`?

File: ui/src/pages/Admin/Upload.tsx [hunk 4]
```diff
@@ -4,11 +4,13 @@ import { Story } from "../../types/index";
 import { UploadStoryWrapper } from "./Edit/UploadStoryWrapper";
 
 export const Upload: React.FC = () => {
+  const startYear = new Date().getFullYear();
```
Comment (AbhiPrasad) [inline ui/src/pages/Admin/Upload.tsx]: We can move this outta the component, no need to compute every render

**Review by AbhiPrasad — APPROVED** [review]
Damn it's kinda wild to see how far you've come.

File: ui/src/pages/Admin/Edit/UploadStory.tsx (hunk 1 above)
Comment (DinuWije) [reply ui/src/pages/Admin/Edit/UploadStory.tsx]: "You can ignore 75% of my PR comments" -- you Tuesday night

File: ui/src/pages/Admin/Edit/UploadStory.tsx (hunk 2 above)
Comment (DinuWije) [reply ui/src/pages/Admin/Edit/UploadStory.tsx:604]: check out #329

File: ui/src/pages/Admin/Edit/UploadStory.tsx (hunk 3 above)
Comment (DinuWije) [reply ui/src/pages/Admin/Edit/UploadStory.tsx]: no real reason... i'll change it to `[]` so it looks cleaner

**Review by nimwijetunga — APPROVED** [review]
lgtm

**Review by hanlinc27 — APPROVED** [review]
giphy (1)

**Pull request merged**